src/read.cpp

```diff
@@ -38,15 +39,27 @@ CharacterVector read_lines_(List sourceSpec, List locale_, std::vector<std::stri
   LocaleInfo locale(locale_);
   Progress progressBar;
 
+  bool chunked = chunk_size != -1;
+
   R_len_t n = (n_max < 0) ? 10000 : n_max;
+
   CharacterVector out(n);
 
   R_len_t i = 0;
+  R_len_t pos = i + 1;
   for (Token t = tokenizer.nextToken(); t.type() != TOKEN_EOF; t = tokenizer.nextToken()) {
     if (progress && (i + 1) % 25000 == 0)
       progressBar.show(tokenizer.progress());
 
-    if (i >= n) {
+    if (chunked && i >= chunk_size) {
+      if (!as<Function>(callback["continue"])()) {
+        break;
+      }
+      out = Rf_xlengthgets(out, i);
```

Comment:
Curious why we use these C APIs instead of the more obvious `CharacterVector.push_back()`? Are the C APIs faster?